common/djangoapps/student/models.py

```diff
@@ -2233,18 +2258,30 @@ def __unicode__(self):
 #### Helper methods for use from python manage.py shell and other classes.
 
 
+def strip_if_string(value):
+    if isinstance(value, basestring):
+        return value.strip()
+    return value
+
+
 def get_user_by_username_or_email(username_or_email):
     """
     Return a User object, looking up by email if username_or_email contains a
     '@', otherwise by username.
 
     Raises:
-        User.DoesNotExist is lookup fails.
+        User.DoesNotExist if no user object can be found, the user was
+        retired, or the user is in the process of being retired.
     """
+    username_or_email = strip_if_string(username_or_email)
     if '@' in username_or_email:
```

Comment:
I have the same concern here as above re: "@" in usernames. I recognize this isn't a new thing, but I'd like to understand it better before perpetuating it.

Reply:
Agreed.  There is another existing-in-our-codebase approach for addressing the problem you bring up, which involves first assuming the identifier is an email, then if no records are found assume that it's a username.  I think that is also problematic if we allow '@' in usernames since there's no uniqueness constraint across the set of all usernames and email adresses combined, so the user with email foo@bar.com is preferentially selected despite the fact that the user with username foo@bar.com may exist.

In new code, I'll just avoid this check altogether and demand explicitly which to attribute check.

Reply:
created https://openedx.atlassian.net/browse/PLAT-2144